src/daemon/ca/keys.rs

```diff
@@ -290,83 +277,74 @@ impl KeyState {
         Ok(RevocationRequest::new(class_name, ki))
     }
 
-    pub fn apply_delta(&mut self, delta: CurrentObjectSetDelta, key_id: KeyIdentifier) {
-        match self {
-            KeyState::Pending(_pending) => panic!("Should never have delta for pending"),
-            KeyState::Active(current) => current.apply_delta(delta),
-            KeyState::RollPending(_pending, current) => current.apply_delta(delta),
-            KeyState::RollNew(new, current) => {
-                if new.key_id() == &key_id {
-                    new.apply_delta(delta)
-                } else {
-                    current.apply_delta(delta)
-                }
-            }
-            KeyState::RollOld(current, old) => {
-                if current.key_id() == &key_id {
-                    current.apply_delta(delta)
-                } else {
-                    old.apply_delta(delta)
-                }
-            }
-        }
-    }
-
     pub fn make_entitlement_events(
         &self,
         rcn: ResourceClassName,
         entitlement: &EntitlementClass,
         base_repo: &RepoInfo,
         name_space: &str,
         signer: &KrillSigner,
-    ) -> KrillResult<Vec<EvtDet>> {
+    ) -> KrillResult<Vec<CaEvtDet>> {
```

Comment:
Maybe extract all the duplicated `let repo = ...; keys_for_requests.push(...)` line blocks into a private nested helper fn?